FWCore/Framework/test/unit_test_outputs/test_deepCall_unscheduled.log

```diff
@@ -109,6 +106,9 @@ ModuleCallingContext state = Running
     ProcessContext: TEST cf65ff121f7d0ed0d094247b80407269
 
 ++++ finished: begin run: stream = 0 run = 1 time = 1000000
+++++ queuing: EventSetup synchronization run: 1 lumi: 1 event: 0
+++++ pre: EventSetup synchronizing run: 1 lumi: 1 event: 0
+++++ post: EventSetup synchronizing run: 1 lumi: 1 event: 0
```

Comment:
Why does order of beginRun and EventSetup synchronization change?

Reply:
First note that the EventSetup signals are for the lumi initialization, not the run initialization. I don't think it technically matters whether they come before or after stream begin run.

This is related to the previous question. Those EventSetup signals fire while ```beginLumiAsync``` is being executed (not while tasks it starts are executed, it's at the top level). We've moved ```beginLumiAsync``` inside the ```then``` part of the new chain so it does not execute while ```handleNextItemAfterMergingRunEntries``` executes at the top level anymore, it is now inside a separate task that it creates. There is intentionally synchronization in there to get stream begin run to execute before global begin lumi in the single threaded case. This preserved long standing behavior in the single threaded case. In the multi threaded case, it is possible for stream end run and the EventSetup signals (and global begin lumi) to all execute concurrently.

I like this new ordering in the single threaded case. It puts the lumi EventSetup signals closer to global begin lumi. Although it was just an accidental side effect of the changes.